bundles/tools.vitruv.domains.java.ui/src/tools/vitruv/domains/java/ui/monitorededitor/JavaMonitoredEditor.xtend

```diff
@@ -126,14 +134,40 @@ class JavaMonitoredEditor extends AbstractMonitoredEditor implements ChangeOpera
 							" was not found although it should exist")
 					}
 				}
-			try {
-				this.virtualModel.propagateChangedState(changedResource, change.oldResourceURI)
+			//TODO This is just a workaround to access the previous resource state.
+			// The correct behavior would be to attach the Java monitor to a state-deriving view and reuse its change deriving functionality.
+			try (val view = getViewForURI(change.oldResourceURI ?: change.newResourceURI)) {
+			    val oldResource = view.rootObjects.head?.eResource
+			    val changeSequence = generateChange(changedResource, oldResource)
```

Comment:
What does this do if `oldResource == null`? Will it fail?

Reply:
You mean in the `generateChange` method? If `oldResource == null` it will call `changeResolutionStrategy.getChangeSequenceForCreated`.

Reply:
I am sorry, forget about my comment. I somehow thought `oldResource` is passed somewhere it might not be allowed to be `null`, but `generateChange` actually has proper handling of that case.